packages/flutter/lib/src/material/tabs.dart

```diff
@@ -1541,6 +1543,7 @@ class _TabBarViewState extends State<TabBarView> {
     return NotificationListener<ScrollNotification>(
       onNotification: _handleScrollNotification,
       child: PageView(
+        key: _pageViewKey,
```

Comment:
Switching the key like this is a little hacky. It often comes with other problems like state-loss in the children because they'll also be re-instantiated.

Reply:
Make sense.